lib/Sema/CSSimplify.cpp

```diff
@@ -2980,6 +2980,16 @@ bool ConstraintSystem::repairFailures(
                        locator);
     }
 
+    // Let's not complain about argument type mismatch if we have a synthesized
+    // wrappedValue initializer, because we already emit a diagnostic for a
+    // type mismatch between the property's type and the wrappedValue type.
+    if (auto CE = dyn_cast<CallExpr>(loc->getAnchor())) {
+      if (CE->isImplicit() && !CE->getArgumentLabels().empty() &&
+          CE->getArgumentLabels().front() == getASTContext().Id_wrappedValue) {
+        break;
```

Comment:
While trying to remove more logic from CSDiag I realized that this is actually not great because it forces fallback to CSDiag. I think we should move this to ArgumentTypeMismatchFailure and diagnose it the same way we do for regular properties e.g.:

```swift
struct S {
  var x: Int = "hello"
}
```

Produces `cannot convert value of type 'String' to specified type 'Int'`

@theblixguy WDYT?

Reply:
I was thinking about a custom diagnostic in this case something like `cannot initialize property of type '<type>' with specified type '<type>'`...

Reply:
Sure, but I think it’s a little redundant to have two separate warnings that basically say the same thing. Perhaps we can change the existing wrappedValue one?

Reply:
What do you mean by existing wrappedValue one?

Reply:
`property type 'Foo' does not match that of the 'wrappedValue' property of its wrapper type 'Bar’`

Reply:
Oh wait, I think I misunderstood. This is for the synthesised wrappedValue init! Yeah, in that case I think it makes sense to introduce a tailored diagnostic.